src/backend/marsha/bbb/api.py

```diff
@@ -570,3 +571,78 @@ def perform_destroy(self, instance):
             )
             document.is_default = True
             document.save()
+
+
+class ClassroomRecordingViewSet(
+    APIViewMixin,
+    ObjectPkMixin,
+    ObjectRelatedMixin,
+    viewsets.GenericViewSet,
+):
+    """
+    Viewset for the API of the ClassroomRecording object.
+    """
+
+    queryset = ClassroomRecording.objects.all()
+    serializer_class = serializers.ClassroomRecordingSerializer
+    filter_backends = [
+        filters.OrderingFilter,
+    ]
+    ordering_fields = ["created_on"]
+    ordering = ["-created_on"]
+
+    permission_classes = []
+
+    def get_permissions(self):
+        """Instantiate and return the list of permissions that this view requires."""
+        if self.action in ["create_vod"]:
+            permission_classes = [
+                core_permissions.IsTokenInstructor
+                | core_permissions.IsTokenAdmin
+                | IsRelatedClassroomPlaylistOrOrganizationAdmin
+            ]
+        else:
+            permission_classes = self.permission_classes
+        return [permission() for permission in permission_classes]
+
+    @action(methods=["post"], detail=True, url_path="create-vod")
+    # pylint: disable=unused-argument
+    def create_vod(self, request, pk=None, classroom_id=None):
```

Comment:
Because, I find it weird to use create : IMO create should use the post data to create a new resource.
This is not what we want, as the data are already in the database.
Also, no recording is created here, but a vod…
WDYT ?

Reply:
I agree with you